### PR TITLE
feat(admin-ui): adds category ui for tree/list, edit, create, delete

### PR DESCRIPTION
**Pull request by riqwan**
**What:**

- Adds a category tree drag and drop component
- Category can be created
- Category can be updated
- Category can be deleted

RESOLVES CORE-1077
RESOLVES CORE-1078
RESOLVES CORE-1200
RESOLVES CORE-1211
RESOLVES CORE-1080
RESOLVES CORE-1079

### CODE REVIEW

File: packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+        const newParent = get(items, path)
+        parentId = newParent.id
+      }
+
+      try {
+        await client.admin.productCategories.update(dragItem.id, {
+          parent_category_id: parentId,
+          rank,
+        })
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:59]: question: Should we add a hook? 🤔

**Review by olivermrbl — APPROVED** [review]
LGTM!

File: packages/admin-ui/ui/src/domain/product-categories/modals/edit-product-category.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  const onSave = async () => {
+    try {
+      await updateCategory({
+        name,
+        handle,
+        is_active: isActive,
+        is_internal: !isPublic,
+      })
+
+      // TODO: check on the BD, when we send update partial children of the category are lost
```
[the remark this answers is not in the document]
Comment (olivermrbl) [reply packages/admin-ui/ui/src/domain/product-categories/modals/edit-product-category.tsx:69]: Great haha! :D

File: packages/admin-ui/ui/src/domain/product-categories/pages/index.tsx [hunk 3]
```diff
@@ -43,8 +88,27 @@ function ProductCategoryPage() {
...
+            )}
           </BodyCard>
+          {isCreateModalVisible && (
+            <CreateProductCategory
+              parentCategory={activeCategory}
+              closeModal={() => {
+                hideCreateModal()
+                setActiveCategory(undefined)
+              }}
+            />
```
[the remark this answers is not in the document]
Comment (olivermrbl) [reply packages/admin-ui/ui/src/domain/product-categories/pages/index.tsx:104]: Got it - thx for explaining @fPolic 💪

File: packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx (hunk 1 above)
Comment (olivermrbl) [reply packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:59]: @riqwan Are we missing a hook? Think this is the only question hindering us from merging this one :)

Comment (fPolic) [reply packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:59]: We have a hook for update but it accepts category id which is unpractical to use in this case

Comment (olivermrbl) [reply packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:59]: Got it, I don't mind leaving it as is, but ideally we would be using hooks for everything.

Comment (fPolic) [reply packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:59]: Agree, currently, we use pass ids to hooks directly to update hooks which has its drawback when the id is dynamically determined and not known during the rendering of the hook, maybe we could write update hooks in the future to pass the id to the mutation function that the hook returns so we have flexibility.

**Pull request merged**